### PR TITLE
fix(Send transaction):Fix user not being to send token due to depreca…

### PR DESCRIPTION
**Pull request by Baksman**
…ted package api

### CODE REVIEW

**Comment by allanclempe** [comment]
first at all, thanks for the contrib @Baksman

File: lib/service/contract_service.dart [hunk 1]
```diff
@@ -8,7 +8,8 @@ typedef TransferEvent = void Function(
 );
 
 abstract class IContractService {
-  Future<Credentials> getCredentials(String privateKey);
+  EthPrivateKey  getCredentials(String privateKey) => EthPrivateKey.fromHex(privateKey);
```
Comment (allanclempe) [inline lib/service/contract_service.dart]: this shouldn't need implementation. consider having the method signature only
`EthPrivateKey getCredentials(String privateKey);`

File: lib/service/contract_service.dart [hunk 2]
```diff
@@ -27,16 +28,18 @@ class ContractService implements IContractService {
   ContractEvent _transferEvent() => contract.event('Transfer');
   ContractFunction _balanceFunction() => contract.function('balanceOf');
   ContractFunction _sendFunction() => contract.function('transfer');
+   @override
+  EthPrivateKey  getCredentials(String privateKey) => EthPrivateKey.fromHex(privateKey);
 
-  @override
-  Future<Credentials> getCredentials(String privateKey) =>
-      client.credentialsFromPrivateKey(privateKey);
+  // @override
```
Comment (allanclempe) [inline lib/service/contract_service.dart]: consider removing commented code.

File: lib/service/contract_service.dart [hunk 3]
```diff
@@ -50,13 +53,15 @@ class ContractService implements IContractService {
     }
 
     try {
+      // TODO: Fix send txn
```
Comment (allanclempe) [inline lib/service/contract_service.dart]: this should be working already, what exactly needs to be fixed?

File: pubspec.yaml [hunk 4]
```diff
@@ -45,6 +46,7 @@ dependencies:
   built_value: ^8.1.4
   built_collection: ^5.1.1
   ed25519_hd_key: ^2.2.0
+  json_rpc_2: ^3.0.1
```
Comment (allanclempe) [inline pubspec.yaml]: is it being used? if not could you please remove it,

**Review by allanclempe — CHANGES_REQUESTED** [review]

**Comment by Baksman** [comment]
> Okay  fixed as requested

**Pull request merged**